src/mips64/cp0.rs

```diff
@@ -77,8 +77,26 @@ impl Cop for Cp0 {
             cop0: self,
         };
         match op.func() {
+            0x00 => {
+                // MFC0
+                match op.rd() {
+                    12 => {
+                        op.cpu.regs[op.rt()] = op.cop0.reg_status;
+                        op.cpu.tight_exit = true;
```

Comment:
I *think* you don't need `tight_exit` here. `tight_exit` is required only if the opcode changes PC either directly or indirectly (eg. can cause an interrupt to be generated). `MFC` shouldn't cause that, while `MTC` can affect interrupt bitmasks in cop0 and thus cause an interrupt (eg: enable an interrupt that was previously disabled).

Reply:
makes sense, I wasn't entirely sure about the intentions behind `tight_exit`